hs-abci-sdk/test/Tendermint/SDK/Test/SimpleStorage.hs

```diff
@@ -0,0 +1,182 @@
+import           Tendermint.SDK.Types.Message     (Msg (..))
+import           Tendermint.SDK.Types.Transaction (PreRoutedTx (..), Tx (..))
+
+--------------------------------------------------------------------------------
+-- Types
+--------------------------------------------------------------------------------
+
+newtype Count = Count Int32 deriving (Eq, Show, Ord, Num, Serialize.Serialize)
+
+data CountKey = CountKey
+
+instance HasCodec Count where
+    encode = Serialize.encode
+    decode = first cs . Serialize.decode
+
+instance BaseApp.RawKey CountKey where
+    rawKey = iso (\_ -> cs countKey) (const CountKey)
+      where
+        countKey :: ByteString
+        countKey = convert . hashWith SHA256 . cs @_ @ByteString $ ("count" :: String)
+
+instance BaseApp.IsKey CountKey "simple_storage" where
+    type Value CountKey "simple_storage" = Count
+
+instance BaseApp.FromQueryData CountKey
+
+instance BaseApp.Queryable Count where
+  type Name Count = "count"
+
+--------------------------------------------------------------------------------
+-- Message Types
+--------------------------------------------------------------------------------
+
+data SimpleStorageMessage =
+  UpdateCount UpdateCountTx
+
+data UpdateCountTx = UpdateCountTx
+  { updateCountTxCount    :: Int32
+  } deriving (Show, Eq, Generic)
+
+--------------------------------------------------------------------------------
+-- Keeper
+--------------------------------------------------------------------------------
+
+storeKey :: BaseApp.StoreKey "simple_storage"
+storeKey = BaseApp.StoreKey "simple_storage"
+
+data SimpleStorage m a where
+    PutCount :: Count -> SimpleStorage m ()
+    GetCount :: SimpleStorage m Count
+
+makeSem ''SimpleStorage
+
+type SimpleStorageEffs = '[SimpleStorage]
+
+updateCount
+  ::  Member SimpleStorage r
+  => Count
+  -> Sem r ()
+updateCount count = putCount count
+
+eval
+  :: forall r.
+     Members '[BaseApp.RawStore, Error BaseApp.AppError] r
+  => forall a. (Sem (SimpleStorage ': r) a -> Sem r a)
+eval = interpret (\case
+  PutCount count -> BaseApp.put storeKey CountKey count
+  GetCount -> fromJust <$> BaseApp.get storeKey CountKey
+  )
+
+--------------------------------------------------------------------------------
+-- Router
+--------------------------------------------------------------------------------
+
+router
+  :: Member SimpleStorage r
+  => Members BaseApp.TxEffs r
+  => PreRoutedTx SimpleStorageMessage
+  -> Sem r ()
+router (PreRoutedTx Tx{txMsg}) =
+  let Msg{msgData} = txMsg
+  in case msgData of
+       UpdateCount UpdateCountTx{updateCountTxCount} ->
+          updateCount (Count updateCountTxCount)
+
+--------------------------------------------------------------------------------
+-- Server
+--------------------------------------------------------------------------------
+
+type CountStoreContents = '[(CountKey, Count)]
+
+type GetMultipliedCount =
+     "multiply"
```

Comment:
Probably mean `multiplier`, as indicated by the tests: https://github.com/f-o-a-m/hs-abci/pull/194/files#diff-f3e7ffe5a8bd3efb979298b272d06c93R46